### PR TITLE
Feature game logic

### PR DESCRIPTION
**Pull request by DejazmachMolla**
For this pull request I

- Created model classes for Player and Game

- Wrote logic for winning, draw, board update, receiving valid entries, switching players and other helpers

- Reverted the cop disable introduced in the user interface branch

- Come up with a fully working game that loops based on interest to continue

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -44,8 +43,6 @@ Lint/ScriptPermission:
   Enabled: false
 Lint/StructNewOverride:
   Enabled: false
-Style/GlobalVars:
-  Enabled: false
```
Comment (lekegitrepo) [inline .rubocop.yml:48]: #### Do not modify linter file; undo any changes.

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
### Request changes :recycle: 

#### Hi @DejazmachMolla :clap: :clap:
#### Good job so far, however, you need to fix some minor issues before this project can be approved.

- **_All game-logic related resources (classes and/or modules) should be kept in the `lib` directory. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone1/lib_dir_exist.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))_**

**_Linters config: Do not make any changes in config files - they represent style guidelines that you share with your team - which is a group of all Microverse students. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/linter_config.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/54883773/))_**

File: .rubocop.yml (hunk 1 above)
Comment (DejazmachMolla) [reply .rubocop.yml:48]: Undone. I think you are looking at an old branch which has already been approved.

**Review by lekegitrepo — APPROVED** [review]
#### This project is approve :heavy_check_mark:
**_Note: Although the linter issue mentioned in the previous review was not addressed, please, do not modify the linter config file in your project_**.

#### _Reviewed by Temitayo Adeleke_.

**Pull request merged**